### PR TITLE
fix: empty destructuring with `var` should be converted to `const`

### LINKED ISSUE #350: `var to let/const` with empty destructuring
The rule `var to let/const` will not transform empty destructuring. 

This is not the code that an average human would write. I noticed this pattern from `babel`'s test case and my [codemod test](https://github.com/unminify-kit/core/blob/caa76de058289a081b6eef1c22b3cf6387fd3ac0/packages/unminify/src/transformations/__tests__/babel-helpers/slicedToArray.spec.ts#L77-L82).

### Input

```js
var arr = [1,2]
var obj = { name: 'John' }

var [] = arr;
var [first, second] = arr
var {} = obj;
var { name } = obj
```

### Actual Output

```js
const arr = [1,2];
const obj = { name: 'John' };

var [] = arr;
const [first, second] = arr;
var {} = obj;
const { name } = obj;
```

### Expected Output

```js
const arr = [1,2];
const obj = { name: 'John' };

const [] = arr;
const [first, second] = arr;
const {} = obj;
const { name } = obj;
```

### PR DESCRIPTION
**Pull request by pionxzh**
closes #350

### CODE REVIEW

**Comment by nene** [comment]
Thanks for bringing up this interesting corner-case. Does Babel actually generate code containing such empty destructuring cases or is it just an odd corner-case that the tests of Babel check?

**Comment by pionxzh** [comment]
It's more like corner cases that Babel is ensuring they are doing it right because they are... well... valid. You usually won't get this unless you do write one.

File: src/transform/let.js [hunk 1]
```diff
@@ -44,6 +44,18 @@ export default function(ast, loggerInstance) {
...
+
+        // Empty destructuring should be converted to const.
+        //
+        //     var [] = foo;
+        //     var {} = foo;
+        //
+        // Mix of empty destructuring and other variable declarations will be handled
+        // by other parts of the transform.
+        if (node.kind === 'var' && node.declarations.every(isEmptyDestructuring)) {
+          node.kind = 'const';
```
Comment (nene) [inline src/transform/let.js:57]: Modifying the `node.kind` field here is definitely the **easy** way to fix this, but it doesn't align with how the rest of this transform is organized. Until now, there have been two distinct phases:

- first gather all the data of all the var-declarations to be modified.
- then go and convert the vars to let/const.

This change breaks this pattern. Now some var-declarations get modified during this first loop of gathering data, which breaks expectations of somebody reading this code in the future.

I'd rather leave these var-s unchanged than do it this way.

Comment (pionxzh) [reply src/transform/let.js:57]: At first, I was trying to do it in the transformation phase, but nothing about empty restructuring will be collected in the "gathering data" stage. It's _empty_. 😢  That's why I ended up putting it here.

Comment (nene) [reply src/transform/let.js:57]: Yeah, it would need to involve some more reorganization. Currently this conversion loop goes through all variables, and takes the var-declarations of them, but as there are no variables here, so such empty declaration will never be reached.

It's definitely not an easy change to make. There's a currently a heavily beaked-in assumption that all variable declarations actually declare at least a single variable.

One possible middle-ground solution would be to separately gather these empty declaration nodes and then separately loop through them in the modification-phase. This way at least the modification and gathering phases will be kept separate, avoiding confusion in that area.

Comment (pionxzh) [reply src/transform/let.js:57]: Good idea. I will try to separate them.
